Improve XVFB detection / timeout logic

## bug

On Travis see the problem sporadically where XVFB would not start
```
> node server.js --port 7076 "--silent"
Your system is missing the dependency: XVFB
Install XVFB and run Cypress again.
Read our documentation on dependencies for more information:
https://on.cypress.io/required-dependencies
If you are using Docker, we provide containers with all required dependencies installed.
----------
Error: Could not start Xvfb.
----------
Platform: linux (Ubuntu Linux - 14.04)
Cypress Version: 2.1.0
npm ERR! Linux 4.14.12-041412-generic
```

It is rare but happens a lot in repos like https://travis-ci.org/cypress-io/cypress-example-recipes

![screen shot 2018-04-20 at 1 24 37 pm](https://user-images.githubusercontent.com/2212006/39064955-3f6ef684-449e-11e8-873d-9ba21ebb026f.png)

--- a/cli/lib/exec/xvfb.js
+++ b/cli/lib/exec/xvfb.js
@@ -7,6 +7,7 @@ const debugXvfb = require('debug')('cypress:xvfb')
 const { throwFormErrorText, errors } = require('../errors')
 
 const xvfb = Promise.promisifyAll(new Xvfb({
+  timeout: 5000, // milliseconds
   onStderrData (data) {
     if (debugXvfb.enabled) {
       debugXvfb(data.toString())

--- a/cli/package.json
+++ b/cli/package.json
@@ -27,7 +27,7 @@
   "types": "types",
   "dependencies": {
     "@cypress/listr-verbose-renderer": "0.4.1",
-    "@cypress/xvfb": "1.1.3",
+    "@cypress/xvfb": "1.2.3",
     "@types/blob-util": "1.3.3",
     "@types/bluebird": "3.5.18",
     "@types/chai": "4.0.8",

